app/auth/signup/email/page.tsx

```diff
@@ -0,0 +1,113 @@
+'use client'
+import { useRef } from 'react'
+import { useForm } from 'react-hook-form'
+import {
+  Box,
+  Flex,
+  FormLabel,
+  FormErrorMessage,
+  FormControl,
+  Input,
+  Heading,
+  useToast
+} from '@chakra-ui/react'
+import { PrimaryButton } from '@/components/button'
+import { signUp } from './action'
+import { useFormResolver, SignUpSchema } from './schema'
+
+export default function SignUp() {
+  const toast = useToast()
+  const isLoading = useRef(false)
```

Comment:
Hmm I've never seen useRef for loading. useRef is usually used for DOM manipulation. I usually use `useBoolean` for loading when I use chakraUI.
https://chakra-ui.com/docs/hooks/use-boolean